docs/rules.rst

```diff
@@ -2,10 +2,140 @@
+
+.. I find this to have excessive spacing before/after sublists. Probably requires CSS to fix?
+
+* Trio
+
+  * `Documentation <https://trio.readthedocs.io/en/stable/reference-core.html#cancellation-and-timeouts>`__
+
+  * :class:`trio.CancelScope`, :func:`trio.move_on_after`, :func:`trio.move_on_at`, :func:`trio.fail_after`, :func:`trio.fail_at`
+
+* AnyIO
+
+  * `Documentation <https://anyio.readthedocs.io/en/stable/cancellation.html>`__
+
+  * :class:`anyio.CancelScope`, :func:`anyio.move_on_after`, :func:`anyio.fail_after``
+
+* asyncio
+
+  * `Documentation <https://docs.python.org/3/library/asyncio-task.html#timeouts>`__
+
+  * :func:`asyncio.timeout`, :func:`asyncio.timeout_at`
+
+.. _taskgroup_nursery:
+
+TaskGroup / Nursery
+-------------------
+
+A collection of child Tasks that can run concurrently.
+
+* Trio
+
+  * `Documentation <https://trio.readthedocs.io/en/stable/reference-core.html#tasks-let-you-do-multiple-things-at-once>`__
+  * :class:`trio.Nursery`, created with :func:`trio.open_nursery`
+* AnyIO
+
+  * `Documentation <https://anyio.readthedocs.io/en/stable/tasks.html>`__
+  * :class:`anyio.abc.TaskGroup`, created with :func:`anyio.create_task_group`.
+* asyncio
+
+  * `Documentation <https://docs.python.org/3/library/asyncio-task.html#asyncio.TaskGroup>`__
+  * :class:`asyncio.TaskGroup` (since python 3.11)
+
+
+.. _cancelled:
+
+Cancelled / CancelledError
+--------------------------
+
+  Handling cancellation is very sensitive, and you generally never want to catch a cancellation exception without letting it propagate to the library.
+
+  * Trio: :class:`trio.Cancelled`. `Documentation <https://trio.readthedocs.io/en/stable/reference-core.html#cancellation-and-timeouts>`__
+  * AnyIO: :func:`anyio.get_cancelled_exc_class`. `Documentation <https://anyio.readthedocs.io/en/stable/cancellation.html>`__
+  * asyncio: :class:`asyncio.CancelledError`. `Documentation <https://docs.python.org/3/library/asyncio-task.html#task-cancellation>`__
+
+
 General rules
 =============
 
+
+
+.. For some reason using :ref:`timeout_context` fails to find the reference, but :ref:`timeout_context <timeout_context>` works. I have no clue why
+
+.. list-table::
+   :widths: 1 18 40
+   :header-rows: 1
+
+   * - Code
+     - Name
+     - Message
+   * - ASYNC100
+     - scope-no-checkpoint
+     - A :ref:`timeout_context <timeout_context>` does not contain any ``await`` statements.  This makes it pointless, as the timeout can only be triggered by a checkpoint. This check also allows ``yield`` statements, since checkpoints can happen in the caller we yield to.
+   * - ASYNC101
+     - yield-in-taskgroup-or-scope
+     - ``yield`` inside a :ref:`TaskGroup/Nursery <taskgroup_nursery>` or :ref:`timeout_context <timeout_context>` is only safe when implementing a context manager - otherwise, it breaks exception handling.
```

Comment:
```suggestion
     - yield-in-cancel-scope
     - ``yield`` inside a :ref:`cancel scope <cancel_scope>` is only safe when implementing a context manager - otherwise, it breaks exception handling.  See PEP 789 for more information.
```
See https://github.com/python/peps/pull/3782 - the common feature of all of these is that they include a cancel scope, either explicitly (trio, anyio) or implicitly (asyncio).

Reply:
I intentionally left out any "see more" stuff, wanting these to be short blurbs and that extra info to be in the individual rules pages. The table already looks really scuffed and crammed, though that's in large part because I haven't gone for modifying the css to have wider pages and be less mobile-friendly.